plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -340,6 +313,62 @@ public List<JdbcColumnHandle> getColumns(ConnectorSession session, JdbcTableHand
         }
     }
 
+    private Optional<JdbcColumnHandle> getColumn(ConnectorSession session, SchemaTableName schemaTableName, Connection connection, ResultSet resultSet)
+            throws SQLException
+    {
+        String columnName = resultSet.getString("COLUMN_NAME");
+        try {
+            return getColumn(session, schemaTableName, connection, resultSet, columnName);
+        }
+        catch (SQLException | RuntimeException e) {
+            throw new TrinoException(JDBC_ERROR, "Failed to resolve column %s in table %s. %s".formatted(columnName, schemaTableName, firstNonNull(e.getMessage(), e)), e);
+        }
+    }
+
+    private Optional<JdbcColumnHandle> getColumn(ConnectorSession session, SchemaTableName schemaTableName, Connection connection, ResultSet resultSet, String columnName)
+            throws SQLException
+    {
+        JdbcTypeHandle typeHandle = getTypeHandle(resultSet);
+        Optional<ColumnMapping> columnMapping = toColumnMapping(session, connection, typeHandle);
+        log.debug("Mapping data type of '%s' column '%s': %s mapped to %s", schemaTableName, columnName, typeHandle, columnMapping);
+        boolean nullable = resultSet.getInt("NULLABLE") != columnNoNulls;
+        // Note: some databases (e.g. SQL Server) do not return column remarks/comment here.
+        Optional<String> comment = Optional.ofNullable(emptyToNull(resultSet.getString("REMARKS")));
+        return columnMapping
+                .map(ColumnMapping::getType)
+                .map(type -> JdbcColumnHandle.builder()
+                        .setColumnName(columnName)
+                        .setJdbcTypeHandle(typeHandle)
+                        .setColumnType(type)
+                        .setNullable(nullable)
+                        .setComment(comment)
+                        .build())
+                .or(() -> handleUnsupportedColumnMapping(session, typeHandle));
+    }
+
+    private static JdbcTypeHandle getTypeHandle(ResultSet resultSet)
+            throws SQLException
+    {
+        return new JdbcTypeHandle(
+                getInteger(resultSet, "DATA_TYPE").orElseThrow(() -> new IllegalStateException("DATA_TYPE is null")),
+                Optional.ofNullable(resultSet.getString("TYPE_NAME")),
+                getInteger(resultSet, "COLUMN_SIZE"),
+                getInteger(resultSet, "DECIMAL_DIGITS"),
+                Optional.empty(),
+                Optional.empty());
+    }
+
+    protected static Optional<JdbcColumnHandle> handleUnsupportedColumnMapping(ConnectorSession session, JdbcTypeHandle typeHandle)
```

Comment:
This method name is incorrect. You don't handle unsupported column type here, you just verified that it was handled in the `toColumnMapping` method.